fe/fe-core/src/main/java/org/apache/doris/statistics/OlapAnalysisTask.java

```diff
@@ -372,12 +396,64 @@ protected String getIndex() {
+    // visited by most queries. To sample this partition may cause the statistics not accurate.
+    // For example, one table has 366 partitions, partition 1 ~ 365 store date for each day of the year from now.
+    // Partition 0 stores all the history data earlier than 1 year. We want to skip sampling partition 0.
+    protected long getSkipPartitionId(List<Partition> partitions) {
+        if (partitions == null || partitions.size() < StatisticsUtil.getPartitionSampleCount()) {
+            return NO_SKIP_TABLET_ID;
+        }
+        PartitionInfo partitionInfo = ((OlapTable) tbl).getPartitionInfo();
+        if (!PartitionType.RANGE.equals(partitionInfo.getType())) {
+            return NO_SKIP_TABLET_ID;
+        }
+        if (partitionInfo.getPartitionColumns().size() != 1) {
+            return NO_SKIP_TABLET_ID;
+        }
+        Column column = partitionInfo.getPartitionColumns().get(0);
+        if (!column.getType().isDateType()) {
+            return NO_SKIP_TABLET_ID;
+        }
+        PartitionKey lowestKey = PartitionKey.createMaxPartitionKey();
+        long lowestPartitionId = -1;
+        for (Partition p : partitions) {
+            RangePartitionItem item = (RangePartitionItem) partitionInfo.getItem(p.getId());
+            Range<PartitionKey> items = item.getItems();
+            if (!items.hasLowerBound()) {
+                lowestPartitionId = p.getId();
+                break;
+            }
+            if (items.lowerEndpoint().compareTo(lowestKey) < 0) {
+                lowestKey = items.lowerEndpoint();
+                lowestPartitionId = p.getId();
+            }
+        }
+        return lowestPartitionId;
+    }
+
+    protected long pickSamplePartition(List<Partition> partitions, List<Long> pickedTabletIds, long skipPartition) {
+        Partition partition = ((OlapTable) tbl).getPartition(skipPartition);
+        long averageRowsPerPartition;
+        if (partition != null) {
+            LOG.info("Going to skip partition {} in table {}", skipPartition, tbl.getName());
+            // If we want to skip the oldest partition, calculate the average rows per partition value without
+            // the oldest partition, otherwise if the oldest partition is very large, we may skip all partitions.
+            // Because we only pick partitions partitionRowCount >= averageRowsPerPartition.
+            Preconditions.checkState(partitions.size() > 1);
+            averageRowsPerPartition = (tbl.getRowCount() - partition.getRowCount()) / (partitions.size() - 1);
+        } else {
+            averageRowsPerPartition = tbl.getRowCount() / partitions.size();
+        }
         long indexId = info.indexId == -1 ? ((OlapTable) tbl).getBaseIndexId() : info.indexId;
         long pickedRows = 0;
         int pickedPartitionCount = 0;
         for (Partition p : partitions) {
+            if (skipPartition == p.getId()) {
+                LOG.info("Partition {} in table {} skipped", skipPartition, tbl.getName());
```

Comment:
debug?

Reply:
I think we can keep it as info. Because there couldn't be too much of this log, at most one log per table, and it's probably useful when debuging.